protobuf/lib/src/protobuf/pb_map.dart

```diff
@@ -39,6 +39,36 @@ class PbMap<K, V> extends MapBase<K, V> {
     _wrappedMap[key] = value;
   }
 
+  @override
+  bool operator ==(other) {
+    if (identical(other, this)) {
+      return true;
+    }
+    if (other is! PbMap) {
+      return false;
+    }
+    if (other.length != length) {
+      return false;
+    }
+    for (final key in keys) {
+      if (!other.containsKey(key)) {
+        return false;
+      }
+    }
+    for (final key in keys) {
+      if (other[key] != this[key]) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  @override
```

Comment:
We should add dartdoc to pbmap explaining how the hash and equals semantics work.